### PR TITLE
More Precise Exception Handling for Item Rename Operation

### PR DESCRIPTION
**Pull request by duke7553**
As per [this](https://docs.microsoft.com/en-us/uwp/api/windows.storage.storagefile.renameasync?view=winrt-19041#Windows_Storage_StorageFile_RenameAsync_System_String_) documentation, we now handle the following cases during the rename operation to provide specific user feedback through the standard content dialog:

- ArgumentException - equivalent of E_INVALIDARG
- PathTooLongException - equivalent of ERROR_FILENAME_EXCED_RANGE
- FileNotFoundException - equivalent of ERROR_FILE_NOT_FOUND

### CODE REVIEW

**Review by tsvietOK — APPROVED** [review]

**Comment by duke7553** [comment]
I may have messed up the resources, so I'll have to revert the merge conflict resolution when I get a chance.

File: Files/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -990,6 +990,23 @@
...
+    <value>The name specified was invalid. Please check the desired item name and try again.</value>
+  </data>
+  <data name="RenameError.NameInvalid.Title" xml:space="preserve">
+    <value>Invalid item name</value>
+  </data>
+  <data name="RenameError.TooLong.Text" xml:space="preserve">
+    <value>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</value>
+  </data>
+  <data name="RenameError.TooLong.Title" xml:space="preserve">
+    <value>Item name was too long</value>
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:1009]:
```suggestion
    <value>Item name was too long</value>
</data>
```

File: Files/MultilingualResources/Files.de-DE.xlf [hunk 2]
```diff
@@ -1183,6 +1183,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.de-DE.xlf:1208]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.es-ES.xlf [hunk 3]
```diff
@@ -1170,6 +1170,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.es-ES.xlf:1195]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.fr-FR.xlf [hunk 4]
```diff
@@ -1176,6 +1176,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.fr-FR.xlf:1201]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.he-IL.xlf [hunk 5]
```diff
@@ -1172,6 +1172,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.he-IL.xlf:1197]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.hi-IN.xlf [hunk 6]
```diff
@@ -1173,6 +1173,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.hi-IN.xlf:1198]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.it-IT.xlf [hunk 7]
```diff
@@ -1172,6 +1172,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.it-IT.xlf:1197]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.ja-JP.xlf [hunk 8]
```diff
@@ -1176,6 +1176,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.ja-JP.xlf:1201]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.nl-NL.xlf [hunk 9]
```diff
@@ -1171,6 +1171,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.nl-NL.xlf:1196]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.or-IN.xlf [hunk 10]
```diff
@@ -1173,6 +1173,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.or-IN.xlf:1198]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.pl-PL.xlf [hunk 11]
```diff
@@ -1174,6 +1174,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.pl-PL.xlf:1199]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.pt-BR.xlf [hunk 12]
```diff
@@ -1172,6 +1172,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.pt-BR.xlf:1197]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.ru-RU.xlf [hunk 13]
```diff
@@ -1170,6 +1170,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.ru-RU.xlf:1195]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.zh-Hans.xlf [hunk 14]
```diff
@@ -1173,6 +1173,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.zh-Hans.xlf:1198]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.uk-UA.xlf [hunk 15]
```diff
@@ -1170,6 +1170,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.uk-UA.xlf:1195]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.tr-TR.xlf [hunk 16]
```diff
@@ -1174,6 +1174,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.tr-TR.xlf:1199]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

File: Files/MultilingualResources/Files.ta.xlf [hunk 17]
```diff
@@ -1175,6 +1175,29 @@
...
+          <source>Invalid item name</source>
+          <target state="new">Invalid item name</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Text" translate="yes" xml:space="preserve">
+          <source>The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</source>
+          <target state="new">The length of the item name specified exceeds the maximum limit. Please check the desired name and try again.</target>
+        </trans-unit>
+        <trans-unit id="RenameError.TooLong.Title" translate="yes" xml:space="preserve">
+          <source>Item name was too long</source>
+          <target state="new">Item name was too long</target>
```
Comment (yaichenbaum) [inline Files/MultilingualResources/Files.ta.xlf:1200]:
```suggestion
          <target state="new">Item name was too long</target>
        </trans-unit>
```

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

**Review by yaichenbaum — DISMISSED** [review]

**Review by yaichenbaum — APPROVED** [review]

**Review by tsvietOK — APPROVED** [review]

**Pull request merged**